client/lib/eventHubReceiver.ts

```diff
@@ -224,7 +224,10 @@ export class EventHubReceiver extends LinkEntity {
         const ehError = translate(receiverError);
         log.error("[%s] An error occurred for Receiver '%s': %O.",
           this._context.connectionId, this.name, ehError);
-        if (this._receiver && !this._receiver.isClosed()) {
+        if (this._receiver && !this._receiver.isClosed() && !ehError.retryable) {
+          log.error("[%s] Since the user did not close the receiver and the error is not " +
+            "retryable, we let the user know about it by calling the user's error handler.",
+            this._context.connectionId);
           this._onError!(ehError);
```

Comment:
@Matejskrbis - This should fix your problem of unnecessary notification of retryable errors. As of today we retry for 150 times at an interval of 15 seconds, which is ~37.5 minutes. If the service goes down for that much time, then it would be a big problem. In most cases, the error should not occur after 5-6 attempts and the receiver will be connected again.

Reply:
That will surly help distinguishing between fatal and retry-able errors. But maybe 50 retries before sending error would be enough?